Amplify/Categories/DataStore/Model/Model.swift

```diff
@@ -26,6 +26,39 @@ public protocol Model: Codable {
     /// undefined behavior.
     var modelName: String { get }
 
-    /// The Model identifier (aka primary key)
-    var id: Identifier { get }
+    /// For internal use only when a model schema is provided
+    /// (i.e. calls from Flutter)
+    func identifier(schema: ModelSchema) -> ModelIdentifierProtocol
+
+    /// Convenience property to access the serialized value of a model identifier
+    var identifier: String { get }
+}
+
+extension Model {
+    public var identifier: String {
+        guard let schema = ModelRegistry.modelSchema(from: modelName) else {
+            preconditionFailure("Schema not found for \(modelName).")
+        }
+        return identifier(schema: schema).stringValue
+    }
+
+    public func identifier(schema modelSchema: ModelSchema) -> ModelIdentifierProtocol {
+        // resolve current instance identifier fields
+        let fields: ModelIdentifierProtocol.Fields = modelSchema.primaryKey.fields.map {
+            guard let value = self[$0.name] as? Persistable else {
+                preconditionFailure("Identifier field named \($0.name) for model \(modelSchema.name) not found.")
```

Comment:
Need to update it as per Brennan's latest PR on preconditionFailure.

Reply:
Brennan's preconditionFailure PR is in `dev-preview` branch, we'll need to update this in the future.